### PR TITLE
Add ability to set vshard options via graphql.

### LINKED ISSUE #174: Allow provide vshard config options
Allow provide vshard options, for example:
- rebalancer_max_receiving

### PR DESCRIPTION
**Pull request by Satbek**
The following vshard options may be modified:
* rebalancer_max_receiving
* collect_lua_garbage
* sync_timeout
* collect_bucket_garbage_interval
* rebalancer_disbalance_threshold

The new GraphQL mutation added:
```GraphQL
edit_vshard_options(
    rebalancer_max_receiving: Int,
    collect_bucket_garbage_interval: Float,
    collect_lua_garbage: Boolean,
    sync_timeout: Float,
    name: String!,
    rebalancer_disbalance_threshold: Float
): VshardGroup!
```

Closes #174

### CODE REVIEW

File: cartridge/vshard-utils.lua [hunk 1]
```diff
@@ -236,13 +291,23 @@ local function get_known_groups()
...
                 bucket_count = vars.default_bucket_count,
                 bootstrapped = false,
+                rebalancer_max_receiving = vars.default_rebalancer_max_receiving,
+                collect_lua_garbage = vars.default_collect_lua_garbage,
+                sync_timeout = vars.default_sync_timeout,
+                collect_bucket_garbage_interval = vars.default_collect_bucket_garbage_interval,
+                rebalancer_disbalance_threshold = vars.rebalancer_disbalance_threshold
             }
         }
     end
```
Comment (rosik) [inline cartridge/vshard-utils.lua]:
```suggestion
     end

for _, g in pairs(vshard_groups) do
    if g.rebalancer_max_receiving ==nil then
        g.rebalancer_max_receiving = vshard_consts.DEFAULT_REBALANCER_MAX_RECEIVING
    end
    -- handle other options the same way
end
```

File: cartridge/vshard-utils.lua [hunk 2]
```diff
@@ -298,6 +363,11 @@ local function get_vshard_config(group_name, conf)
 
     return {
         bucket_count = vshard_groups[group_name].bucket_count,
+        rebalancer_max_receiving = vshard_groups[group_name].rebalancer_max_receiving,
+        collect_lua_garbage = vshard_groups[group_name].collect_lua_garbage,
+        sync_timeout = vshard_groups[group_name].sync_timeout,
+        collect_bucket_garbage_interval = vshard_groups[group_name].collect_bucket_garbage_interval,
+        rebalancer_disbalance_threshold = vshard_groups[group_name].rebalancer_disbalance_threshold,
```
Comment (rosik) [inline cartridge/vshard-utils.lua]: this is ok

File: cartridge/vshard-utils.lua [hunk 3]
```diff
@@ -345,6 +415,34 @@ local function can_bootstrap()
...
+        name = 'string',
+        rebalancer_max_receiving = '?number',
+        collect_lua_garbage = '?boolean',
+        sync_timeout = '?number',
+        collect_bucket_garbage_interval = '?number',
+        rebalancer_disbalance_threshold = '?number'
+    })
+
+    local patch = {
+        vshard_groups = confapplier.get_deepcopy('vshard_groups')
```
Comment (rosik) [inline cartridge/vshard-utils.lua]: It should `get_deepcopy('vshard')` too

File: cartridge/webui/api-vshard.lua [hunk 4]
```diff
@@ -110,6 +134,21 @@ local function init(graphql)
         kind = gql_types.list(gql_type_vsgroup.nonNull).nonNull,
         callback = module_name .. '.get_vshard_groups',
     })
+
+    graphql.add_mutation({
+        prefix = 'cluster',
+        name = 'edit_vshard_group_params',
```
Comment (rosik) [inline cartridge/webui/api-vshard.lua]: edit_vshard_options

File: test/integration/test_vshard_options.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python3
+
+import pytest
+import logging
+from conftest import Server
+
+init_script = 'srv_vshard_options.lua'
```
Comment (rosik) [inline test/integration/test_vshard_options.py]: Add this test to `test_multisharding_one` and `_two`

File: test/integration/test_multisharding_one.py [hunk 6]
```diff
@@ -93,3 +150,49 @@ def test_router_role(cluster):
...
+                'collect_bucket_garbage_interval': 42.24,
+                'collect_lua_garbage': True,
+                'rebalancer_disbalance_threshold': 14,
+                'rebalancer_max_receiving': 42,
+                'sync_timeout': 24,
+                'name': 'default',
+                'bucket_count': 3000,
+                'bootstrapped': True,
+            }
+        ]
```
Comment (rosik) [inline test/integration/test_multisharding_one.py]: Extend this test: call `edit_vshard_group(sync_timeout = 25, rebalancer_max_receiving = None)`

Check that sync_timeout is updated, other values should remain the same

File: test/integration/test_multisharding_two.py [hunk 7]
```diff
@@ -157,3 +160,91 @@ def test_router_role(cluster):
...
+               'rebalancer_disbalance_threshold': 14,
+               'rebalancer_max_receiving': 42,
+               'sync_timeout': 24,
+               'name': 'hot',
+               'bucket_count': 30000,
+               'bootstrapped': True,
+           }]
+
+
+def test_set_vshard_options_negative(cluster):
```
Comment (rosik) [inline test/integration/test_multisharding_two.py]: No need to check it twice

File: test/integration/test_multisharding_two.py [hunk 8]
```diff
@@ -157,3 +160,91 @@ def test_router_role(cluster):
         'hot': 'bbbbbbbb-bbbb-4000-b000-000000000002',
         'cold': 'cccccccc-cccc-4000-b000-000000000002'
     }
+
+
+def test_set_vshard_options_positive(cluster):
+    obj = edit_vshard_group(cluster, group = 'hot',
```
Comment (rosik) [inline test/integration/test_multisharding_two.py]: No need to check all the options again.

The main test here is that each group can be edited separately.

File: test/unit/test_vshard_config.lua [hunk 9]
```diff
@@ -146,8 +227,8 @@ topology:
       vshard_group: some
       roles: {}
 vshard:
-  bucket_count: 1
-  bootstrapped: false
+    bucket_count: 1
+    bootstrapped: false
```
Comment (rosik) [inline test/unit/test_vshard_config.lua]: Excess indent

**Comment by Satbek** [comment]
Additional changes:
* edit_vshard_group request will return vshard_group
* if in edit_vshard_group param was set to None then this param should not be changed

**Pull request merged**